streaming/consumer.cc

```diff
@@ -70,6 +79,13 @@ std::function<future<> (flat_mutation_reader)> make_streaming_consumer(sstring o
                 });
             });
             co_return co_await consumer(std::move(reader));
+        } catch (...) {
+            ex = std::current_exception();
+        }
+        if (ex) {
+            co_await reader.close();
```

Comment:
The `reader` is moved into `consumer()` inside try block, so if the `consumer()` throws won't this line close moved reader?

Reply:
it's safe to close the reader after it was std::moved
it's left in a valid state where close would be a noop.
See https://github.com/scylladb/scylla/blob/fa29b79c200d81de6e35685c5b70ff6ae278d12a/flat_mutation_reader.hh#L468-L482